🔥🪞 Fix relation prediction with inverses

mberr opened the pull request:
Fix #1303

mberr committed:
add is_inverse method

mberr committed:
filter scores for inverse relations

cthoyt commented on src/pykeen/inverse.py:
```diff
@@ -61,6 +65,10 @@ def invert_(self, batch: torch.LongTensor, index: int = 1) -> torch.LongTensor:
         return batch
 
+    # docstr-coverage: inherited
+    def is_inverse(self, ids: torch.LongTensor) -> torch.BoolTensor:  # noqa: D102
```
somewhere there is also the inverse index generator function, should these be coupled?

mberr replied:
This is exactly what this class is for 😉 

The full inversion combines `_map` and `invert_` defined in this concrete class, `DefaultRelationInverter` ,  in `map` in the base class, `RelationInverter`.

mberr replied:
This PR is mostly a fix for the prediction dataframe; there are still some inconsistencies to fix (e.g. that inverse relations are stored in a dataset instead of solely encapsulated in a model; that tf.relation_to_id contains only the relation ids *before* remapping, etc.), cf. this long-standing PR: https://github.com/pykeen/pykeen/pull/752

mberr committed:
Pin Sphinx < 7 until sphinx_rtd_theme is updated

cf. https://www.sphinx-doc.org/en/master/changes.html#release-7-0-0-released-apr-29-2023
